lib/src/component/hooks.dart

```diff
@@ -474,10 +474,11 @@ void useLayoutEffect(dynamic Function() sideEffect, [List<Object?>? dependencies
 ///
 /// UiFactory<FancyInputProps> FancyInput = uiForwardRef(
 ///   (props, ref) {
```

Comment:
Since we're using a ref declared in props instead of the forwarded ref, we probably should just make this `uiFunction`

```suggestion
/// UiFactory<FancyInputProps> FancyInput = uiFunction(
///   (props) {
```

Reply:
Actually, the ref passed to a component isn't available in `props.ref` and so I believe you need to use forwardRef to use `useImperativeHandle` (see React docs [here](https://react.dev/reference/react/useImperativeHandle#:~:text=The%20ref%20you%20received%20as%20the%20second%20argument%20from%20the%20forwardRef%20render%20function.))